dietify-frontend/src/components/Product/Product.jsx

```diff
@@ -0,0 +1,201 @@
+import React, { useState } from "react";
+import { useProduct } from "../../contexts";
+import { actions } from "../../reducers";
+import ProductListing from "./ProductListing";
+import {
+  getFilteredData,
+  getSearchedData,
+  getSortedData,
+  getPriceRangeData,
+} from "./Filter";
+import { PrimaryButton } from "./ProductListing";
+import "./Product.css";
+
+function Product({ loading }) {
+  const {
+    state: {
+      sortBy,
+      showOutOfStock,
+      showFastDeliveryOnly,
+      productList,
+      search,
+      priceRange,
+    },
+    dispatch,
+  } = useProduct();
+  const [showFilter, setShowFilter] = useState(false);
+
+  const sortPriceLowToHigh = () => {
+    dispatch({ type: actions.PRICE_LOW_TO_HIGH });
+  };
+
+  const sortPriceHighToLow = () => {
+    dispatch({ type: actions.PRICE_HIGH_TO_LOW });
+  };
+
+  const filterOutOfStock = () => {
+    dispatch({ type: actions.FILTER_OUT_OF_STOCK });
+  };
+
+  const filterFastDelivery = () => {
+    dispatch({ type: actions.FILTER_FAST_DELIVERY });
+  };
+
+  const clearAllFilters = () => {
+    dispatch({ type: actions.CLEAR_ALL_FILTERS });
+  };
+
+  const searchProduct = (search) => {
+    dispatch({ type: actions.UPDATE_SEARCH_TEXT, payload: search });
+  };
+
+  const handleSliderChange = (e) => {
+    dispatch({
+      type: actions.UPDATE_PRICE_RANGE,
+      payload: { value: e.target.value },
+    });
+  };
+
+  const sortedData = getSortedData(sortBy, productList);
+  const filteredData = getFilteredData(
+    showOutOfStock,
+    showFastDeliveryOnly,
+    sortedData
+  );
+  const priceRangeData = getPriceRangeData(priceRange, filteredData);
+  const searchedData = getSearchedData(search, priceRangeData);
```

Comment:
If we send the arguments as an object it will be easier to access at the function level 
no need to concern about sequence of arguments

Reply:
Done